model-optimizer/extensions/front/kaldi/tdnn_component_replacer.py

```diff
@@ -0,0 +1,104 @@
+from mo.ops.memoryoffset import MemoryOffset
+
+
+class TdnnComponentReplacer(FrontReplacementPattern):
+    '''
+    Expand TdnnComponent into MemoryOffsets, Concat and FullyConected nodes
+
+    BEFORE:
+                          placeholder
+                              |
+                      TdnnComponent('time_offsets': t1, t2,... tk)
+                              |
+    _______________________________________________________________
+
+    AFTER:
+                          placeholder
+            __________________|___________________________
+           /                  |              \            \
+   MemoryOffset(t1)     MemoryOffset(t2)    ...    MemoryOffset(tk)
+          \_____________ _____|______________/____________/
+                           Concat
+                             |
+                        FullyConnected
+                             |
+    '''
+    enabled = True
+    run_not_recursively = True
+
+    def run_after(self):
+        from extensions.front.restore_ports import RestorePorts
+        return [RestorePorts]
+
+    def run_before(self):
+        from extensions.front.kaldi.memory_offset_adjustment import MemoryOffsetAdjustment
+        return [MemoryOffsetAdjustment]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for node in graph.get_op_nodes(op='tdnncomponent'):
+            self.replace_tdnn(graph, node)
+
+    def replace_tdnn(self, graph: Graph, tdnn_node: Node):
+        tdnn_name = tdnn_node.soft_get('name', tdnn_node.id)
+
+        concat_node = Concat(graph, {'axis': 1}).create_node()
+        rename_nodes([(tdnn_node, tdnn_name + '/to_be_removed'), (concat_node, tdnn_name)])
+
+        for i, t in enumerate(tdnn_node['time_offsets']):
+            concat_node.add_input_port(i)
+            if t != 0:
+                memory_name = tdnn_name + '_memoryoffset_' + str(abs(t))
+                memoryoffset_node = MemoryOffset(graph, {'name': memory_name, 't': t,
+                                                         'pair_name': memory_name + '_out',
+                                                         'has_default': False, 'splitted': False}).create_node()
+
+                tdnn_node.in_port(0).get_source().connect(memoryoffset_node.in_port(0))
+                memoryoffset_node.out_port(0).connect(concat_node.in_port(i))
+            else:
+                # 0 time delay is not allowed in IE, it's meaningless
+                # if time offset is 0 then connect input of tdnncomponent directly to Concat without memoryoffset
+                tdnn_node.in_port(0).get_source().connect(concat_node.in_port(i))
+
+        weights = tdnn_node['weights']
+        bias_term = False
+        if tdnn_node.has_valid('biases') and tdnn_node['biases'] is not None:
+            bias_term = True
+            biases = tdnn_node['biases']
+            assert len(biases) == weights.shape[0]
+
+        fc_node = FullyConnected(graph, {'name': tdnn_name + '_fc',
```

Comment:
Use `create_op_with_const_inputs` method to create `fc_node`.

Reply:
Agree. Changed.